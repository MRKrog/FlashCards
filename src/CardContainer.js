import React, { Component } from 'react';
import Card from './Card.js'

class CardContainer extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const cards = this.props.allCards;

    const cardList = cards.map((card) =>
      <Card singleCard={card} key={card.id}/>
    )

    return (
      <div>{cardList}</div>
    )
  }

}


export default CardContainer;
