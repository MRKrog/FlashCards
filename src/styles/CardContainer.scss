
.CardContainer {
  display: flex;
  justify-content: center;
  padding: 2em;

  @media only screen and (max-width: 600px) {
    padding: 0;
  }
}
