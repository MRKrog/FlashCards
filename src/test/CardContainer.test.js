import React from "react";
import CardContainer from "../CardContainer";
import MockData from '../mockData';
import { shallow } from "enzyme";

const cards = MockData.cards;


describe("CardContainer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CardContainer singleCard={card} />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

});
