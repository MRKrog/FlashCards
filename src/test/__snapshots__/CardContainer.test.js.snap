// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot when all data is passed correctly 1`] = `
<div>
  <Card
    key="101"
    singleCard={
      Object {
        "answers": Array [
          "Home Tool Markup Language",
          "Hyperlinks and Text Markup Language",
          "Hyper Text Material Language",
          "Hyper Text Markup Language",
        ],
        "category": "html",
        "correctAnswer": "Hyper Text Markup Language",
        "id": 101,
        "question": "What does HTML stand for?",
      }
    }
  />
  <Card
    key="102"
    singleCard={
      Object {
        "answers": Array [
          "<*-- A SAMPLE COMMENT --*>",
          "<?-- A SAMPLE COMMENT -->",
          "<!-- A SAMPLE COMMENT -->",
          "<-- A SAMPLE COMMENT --!>",
        ],
        "category": "html",
        "correctAnswer": "<!-- A SAMPLE COMMENT -->",
        "id": 102,
        "question": "How do you insert a comment in HTML?",
      }
    }
  />
  <Card
    key="103"
    singleCard={
      Object {
        "answers": Array [
          "&copy;",
          "&copyRight;",
          "&copy!",
          "&copySymbol;",
        ],
        "category": "html",
        "correctAnswer": "&copy;",
        "id": 103,
        "question": "How do you insert a copyright symbol on a browser page?",
      }
    }
  />
  <Card
    key="104"
    singleCard={
      Object {
        "answers": Array [
          "Miscrosoft",
          "Firefox",
          "The World Wide Web Consortium",
          "Internet Explorer",
        ],
        "category": "html",
        "correctAnswer": "The World Wide Web Consortium",
        "id": 104,
        "question": "Who is making the Web standards?",
      }
    }
  />
  <Card
    key="105"
    singleCard={
      Object {
        "answers": Array [
          "<h1>",
          "<h4>",
          "<head>",
          "<header>",
        ],
        "category": "html",
        "correctAnswer": "<h1>",
        "id": 105,
        "question": "What is the largest HTML element heading tag?",
      }
    }
  />
  <Card
    key="106"
    singleCard={
      Object {
        "answers": Array [
          "<a url='http://www.w3schools.com'>google.com</a>",
          "<a link='http://www.w3schools.com'>google.com</a>",
          "<a script='http://www.w3schools.com'>google.com</a>",
          "<a href='http://www.w3schools.com'>google.com</a>",
        ],
        "category": "html",
        "correctAnswer": "<a href='http://www.w3schools.com'>google.com</a>",
        "id": 106,
        "question": "How do you create an anchor tag that creates a hyperlink?",
      }
    }
  />
  <Card
    key="107"
    singleCard={
      Object {
        "answers": Array [
          "target=\\"_blank\\"",
          "newTab=\\"_blank\\"",
          "tab=\\"_new\\"",
          "target=\\"_tab\\"",
        ],
        "category": "html",
        "correctAnswer": "target=\\"_blank\\"",
        "id": 107,
        "question": "Which HTML attribute can open a new tab in a browser?",
      }
    }
  />
  <Card
    key="108"
    singleCard={
      Object {
        "answers": Array [
          "<ul>",
          "<li>",
          "<ol>",
          "<order>",
        ],
        "category": "html",
        "correctAnswer": "<ol>",
        "id": 108,
        "question": "Which element makes a numbered list?",
      }
    }
  />
  <Card
    key="109"
    singleCard={
      Object {
        "answers": Array [
          "<img src=\\"image.jpg\\">",
          "<img link=\\"image.jpg\\">",
          "<image src=\\"image.jpg\\">",
          "<image link=\\"image.jpg\\">",
        ],
        "category": "html",
        "correctAnswer": "<img src=\\"image.jpg\\">",
        "id": 109,
        "question": "What is the correct HTML for inserting an image?",
      }
    }
  />
  <Card
    key="110"
    singleCard={
      Object {
        "answers": Array [
          "<header>",
          "<title>",
          "<head>",
          "<section>",
        ],
        "category": "html",
        "correctAnswer": "<title>",
        "id": 110,
        "question": "Which HTML element defines the title of a document?",
      }
    }
  />
  <Card
    key="201"
    singleCard={
      Object {
        "answers": Array [
          "Computer Style Sheets",
          "Creative Style Sheets",
          "Cascading Stylist Sheets",
          "Cascading Style Sheets",
        ],
        "category": "css",
        "correctAnswer": "Cascading Style Sheets",
        "id": 201,
        "question": "What does CSS stand for?",
      }
    }
  />
  <Card
    key="202"
    singleCard={
      Object {
        "answers": Array [
          "<footer>",
          "<head>",
          "<body>",
          "<section>",
        ],
        "category": "css",
        "correctAnswer": "<head>",
        "id": 202,
        "question": "Which tag is the correct place to refer to an external style sheet?",
      }
    }
  />
  <Card
    key="203"
    singleCard={
      Object {
        "answers": Array [
          "<css>",
          "<script>",
          "<head>",
          "<style>",
        ],
        "category": "css",
        "correctAnswer": "<style>",
        "id": 203,
        "question": "Which HTML tag is used to define an internal style sheet?",
      }
    }
  />
  <Card
    key="204"
    singleCard={
      Object {
        "answers": Array [
          "body:color=black;",
          "{body;color:black;}",
          "body{color: black;}",
          "{body:color=black;}",
        ],
        "category": "css",
        "correctAnswer": "body{color: black;}",
        "id": 204,
        "question": "Which is the correct CSS syntax?",
      }
    }
  />
  <Card
    key="205"
    singleCard={
      Object {
        "answers": Array [
          "// this is a comment",
          "*this is a comment*",
          "/* this is a comment */",
          "// this is a comment //",
        ],
        "category": "css",
        "correctAnswer": "/* this is a comment */",
        "id": 205,
        "question": "How do you insert a comment in a CSS file?",
      }
    }
  />
  <Card
    key="206"
    singleCard={
      Object {
        "answers": Array [
          "background-color",
          "color",
          "bgcolor",
          "back-color",
        ],
        "category": "css",
        "correctAnswer": "background-color",
        "id": 206,
        "question": "Which property is used to change the background color?",
      }
    }
  />
  <Card
    key="207"
    singleCard={
      Object {
        "answers": Array [
          "fgcolor",
          "color",
          "text-color",
          "background-text",
        ],
        "category": "css",
        "correctAnswer": "color",
        "id": 207,
        "question": "Which CSS property is used to change the text color of an element?",
      }
    }
  />
  <Card
    key="208"
    singleCard={
      Object {
        "answers": Array [
          "text-style",
          "text-size",
          "font-size",
          "font-style",
        ],
        "category": "css",
        "correctAnswer": "font-size",
        "id": 208,
        "question": "Which CSS property controls the text size?",
      }
    }
  />
  <Card
    key="209"
    singleCard={
      Object {
        "answers": Array [
          "a {text-decoration:no-underline;}",
          "a {text-decoration:none;}",
          "a {decoration:no-underline;}",
          "a {underline:none;}",
        ],
        "category": "css",
        "correctAnswer": "a {text-decoration:none;}",
        "id": 209,
        "question": "How do you display hyperlinks without an underline?",
      }
    }
  />
  <Card
    key="210"
    singleCard={
      Object {
        "answers": Array [
          "font-weight",
          "font-family",
          "font-style",
          "font-text",
        ],
        "category": "css",
        "correctAnswer": "font-weight",
        "id": 210,
        "question": "Which property is used to change the font of an element?",
      }
    }
  />
  <Card
    key="301"
    singleCard={
      Object {
        "answers": Array [
          "<script>",
          "<javascript>",
          "<html>",
          "<css>",
        ],
        "category": "javascript",
        "correctAnswer": "<script>",
        "id": 301,
        "question": "Inside which HTML element do we put JavaScript into?",
      }
    }
  />
  <Card
    key="302"
    singleCard={
      Object {
        "answers": Array [
          "msgBox(\\"Hello World\\");",
          "alert(\\"Hello World\\");",
          "console.log(\\"Hello World\\");",
          "alertBox(\\"Hello World\\");",
        ],
        "category": "javascript",
        "correctAnswer": "alert(\\"Hello World\\");",
        "id": 302,
        "question": "How do you write \\"Hello World\\" in an alert box?",
      }
    }
  />
  <Card
    key="303"
    singleCard={
      Object {
        "answers": Array [
          "function = myFunction()",
          "function:myFunction()",
          "function myFunction()",
          "function()",
        ],
        "category": "javascript",
        "correctAnswer": "function myFunction()",
        "id": 303,
        "question": "How do you create a function in JavaScript?",
      }
    }
  />
  <Card
    key="304"
    singleCard={
      Object {
        "answers": Array [
          "new exampleFunction()",
          "run exampleFunction",
          "exampleFunction",
          "exampleFunction ()",
        ],
        "category": "javascript",
        "correctAnswer": "exampleFunction()",
        "id": 304,
        "question": "How do you call a function named exampleFunction?",
      }
    }
  />
  <Card
    key="305"
    singleCard={
      Object {
        "answers": Array [
          "if (i == 5)",
          "if i == 5 then",
          "if i = 5 then",
          "if i = 5",
        ],
        "category": "javascript",
        "correctAnswer": "if (i == 5)",
        "id": 305,
        "question": "How to write an IF statement in JavaScript?",
      }
    }
  />
  <Card
    key="306"
    singleCard={
      Object {
        "answers": Array [
          "while (i <= 5; i++)",
          "while (i = 0; i <= 5)",
          "while i = 1 to 5",
          "while (i <= 5)",
        ],
        "category": "javascript",
        "correctAnswer": "while (i <= 5)",
        "id": 306,
        "question": "How does a WHILE loop start?",
      }
    }
  />
  <Card
    key="307"
    singleCard={
      Object {
        "answers": Array [
          "for (i <= 5; i++)",
          "for i = 1 to 5",
          "for (i = 0; i <= 5; i++)",
          "for (i = 0; i <= 5)",
        ],
        "category": "javascript",
        "correctAnswer": "for (i = 0; i <= 5; i++)",
        "id": 307,
        "question": "How does a FOR loop start?",
      }
    }
  />
  <Card
    key="308"
    singleCard={
      Object {
        "answers": Array [
          "'This is a comment",
          "//This is a comment",
          "/*This is a comment*/",
          "<!--This is a comment-->",
        ],
        "category": "javascript",
        "correctAnswer": "//This is a comment",
        "id": 308,
        "question": "How can you add a comment in a JavaScript?",
      }
    }
  />
  <Card
    key="309"
    singleCard={
      Object {
        "answers": Array [
          "onhover",
          "onclick",
          "onmouseover",
          "event",
        ],
        "category": "javascript",
        "correctAnswer": "onclick",
        "id": 309,
        "question": "Which event occurs when the user clicks on an HTML element?",
      }
    }
  />
  <Card
    key="310"
    singleCard={
      Object {
        "answers": Array [
          "v exampleVar",
          "variable exampleVar",
          "var exampleVar",
          "new exampleVar",
        ],
        "category": "javascript",
        "correctAnswer": "var exampleVar",
        "id": 310,
        "question": "How do you declare a JavaScript variable?",
      }
    }
  />
</div>
`;
